commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java

```diff
@@ -1,69 +1,135 @@
+import eu.europeana.api.commons.web.model.vocabulary.UserRoles;
+
+public abstract class BaseAuthorizationService implements AuthorizationService {
+
+    RsaVerifier signatureVerifier;
+	    
+    protected RsaVerifier getSignatureVerifier() {
+        if(signatureVerifier == null)
+            signatureVerifier = new RsaVerifier(getSignatureKey()); 
+	return signatureVerifier;
+    }
+
+    @Override
+    public void authorizeReadAccess(HttpServletRequest request) throws ApplicationAuthenticationException {
+    
+	String wsKey; 
+	//check and verify jwt token
+	try {
+	    wsKey = OAuthUtils.extractApiKeyFromJwtToken(request, getSignatureVerifier());
+	    if(wsKey != null) {
+        	return;//apikey is valid if the JWT Token is verified
+	    }
+	} catch (ApiKeyExtractionException e) {
+	    throw new ApplicationAuthenticationException(I18nConstants.INVALID_APIKEY, I18nConstants.INVALID_APIKEY,
+		    new String[] { e.getMessage() }, HttpStatus.UNAUTHORIZED, e);
+	}
+	
+	//extract api key with other methods
+	try {
+		wsKey = OAuthUtils.extractApiKey(request);
+	} catch (ApiKeyExtractionException e) {
+	    throw new ApplicationAuthenticationException(I18nConstants.INVALID_APIKEY, I18nConstants.INVALID_APIKEY,
+		    new String[] { e.getMessage() }, HttpStatus.UNAUTHORIZED, e);
+	}
+	
+	//check if null
+	if (wsKey == null)
+	    throw new ApplicationAuthenticationException(I18nConstants.MISSING_APIKEY, I18nConstants.MISSING_APIKEY,
+		    null, HttpStatus.UNAUTHORIZED, null);
+	//check if empty
+	if (StringUtils.isEmpty(wsKey))
+	    throw new ApplicationAuthenticationException(I18nConstants.EMPTY_APIKEY, I18nConstants.EMPTY_APIKEY, null);
+	//validate api key
+	if (!wsKey.equals("apidemo")) {
+	    try {
+		getClientDetailsService().loadClientByClientId(wsKey);
+	    } catch (Exception e) {
+		throw new ApplicationAuthenticationException(I18nConstants.INVALID_APIKEY, I18nConstants.INVALID_APIKEY,
+			new String[] { wsKey });
+	    }
+	}
+    }
+
+     
+    /* (non-Javadoc)
+     * @see eu.europeana.api.commons.service.authorization.AuthorizationService#authorizeWriteAccess(java.util.List, java.lang.String, java.lang.String)
+     */
+    public boolean authorizeWriteAccess(List<? extends Authentication> authenticationList, String operation) 
+	    throws ApplicationAuthenticationException {
+	    
+	return checkPermissions(authenticationList, getAuthorizationApiName(), operation);
+    }
+    
+    /**
+     * This method verifies write access rights for particular api and operation
+     * @param authenticationList The list of authentications extracted from the JWT token
+     * @param api The current api name
+     * @param operation The name of current operation
+     * @return true if authenticated, false otherwise
+     * @throws ApplicationAuthenticationException
+     */
+    public static boolean checkPermissions(List<? extends Authentication> authenticationList, String api, String operation) 
```

Comment:
doesn't nneed to be public, protected is enough

Reply:
changed to protected

Reply:
the method must not be static